src/cluster.py

```diff
@@ -168,11 +179,21 @@ def _create_directory(self, path: str, mode: int) -> None:
         os.chmod(path, mode)
         self._change_owner(path)
 
-    def _get_postgresql_version(self) -> str:
+    def _create_user_home_directory(self) -> None:
+        """Creates the user home directory for the snap_daemon user.
+
+        This is needed due to https://bugs.launchpad.net/snapd/+bug/2011581.
+        """
+        subprocess.run("mkdir -p /home/snap_daemon".split())
+        subprocess.run("chown snap_daemon:snap_daemon /home/snap_daemon".split())
+        subprocess.run("usermod -d /home/snap_daemon snap_daemon".split())
+
+    def get_postgresql_version(self) -> str:
         """Return the PostgreSQL version from the system."""
-        package = DebianPackage.from_system("postgresql")
-        # Remove the Ubuntu revision from the version.
-        return str(package.version).split("+")[0]
+        client = snap.SnapClient()
+        for snp in client.get_installed_snaps():
+            if snp["name"] == POSTGRESQL_SNAP_NAME:
+                return snp["version"]
```

Comment:
`postgres` is currently not exposed as a snap app and I'm not sure if it would be a good idea to expose it. Alternatively we can use another utility for that (`pg_ctl`?). Would this be OK to stay as is for the time being?

Reply:
yes, `pg_ctl` good enough